src/NServiceBus.AcceptanceTests/Recoverability/Retries/When_message_goes_through_all_retries.cs

```diff
@@ -0,0 +1,176 @@
+﻿namespace NServiceBus.AcceptanceTests.Recoverability.Retries
+{
+    using System;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using EndpointTemplates;
+    using Features;
+    using NServiceBus.Config;
+    using NServiceBus.Pipeline;
+    using NUnit.Framework;
+    using ScenarioDescriptors;
+
+    public class When_message_goes_through_all_retries : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_not_send_outgoing_messages_using_batched_dispatch()
```

Comment:
for batched dispatch with multi-queue transactions, it would actually make sense to throw an exception in the `ITransportReceiveContext` stage, because that's where batched messages have been sent.

It seems like we'd rather have tests using single-queue transactions to verify that we "rollback" outgoing messages on exceptions before the `ITransportReceiveContext` stage (and dispatch them afterwards).